pkg/fanal/analyzer/imgconf/dockerfile/dockerfile.go

```diff
@@ -119,7 +117,7 @@ func imageConfigToDockerfile(cfg *v1.ConfigFile) []byte {
 		dockerfile.WriteString(strings.TrimSpace(createdBy) + "\n")
 	}
 
-	if !userFound && cfg.Config.User != "" {
+	if cfg.Config.User != "" {
```

Comment:
nit:
```suggestion
	// The user can be changed from the config file or with the `--user` flag (for docker CLI), so we need to add this user to avoid incorrect user detection
	if cfg.Config.User != "" {
```